lib/index.js

```diff
@@ -19,23 +31,47 @@ module.exports = (referenceDir, config, { watch, fingerprint, sourcemaps, compac
 		fingerprint,
 		exitOnError: !watch
 	});
-	let watcher = watch && makeWatcher(config.watchDirs, referenceDir,
-			assetManager.resolvePath);
 	let browsers = browserslist.findConfig(referenceDir) || {};
 
 	let plugins = Object.assign({}, DEFAULTS.plugins, config.plugins);
+	let buckets = {
+		static: [],
+		scripts: [],
+		styles: [],
+		markup: []
+	};
 	Object.keys(plugins).forEach(type => {
 		let pluginConfig = config[type];
 		if(!pluginConfig) {
 			return;
 		}
 
-		let plugin = plugins[type];
+		let bucket = plugins[type].bucket;
+		let plugin = plugins[type].package;
 		if(!plugin.call) {
 			plugin = load(plugin);
 		}
-		plugin(pluginConfig, assetManager, { watcher, browsers, sourcemaps, compact });
+		let initializedPlugin = plugin(pluginConfig, assetManager,
+				{ browsers, sourcemaps, compact });
+		buckets[bucket].push(initializedPlugin);
```

Comment:
`initializedPlugin` seems like a weird term somehow; it doesn't strike me as very descriptive? Also, within `buildAll` below, it's just called `plugin` AFAICT.

Reply:
If I understand you correctly, you suggest:

```
plugin = plugin(pluginConfig, assetManager, { browsers, sourcemaps, compact })
```

Right? I think this is not a good idea. Reassigning a variable from a function to the applied function is weird. What is the difference between the plugin before and after calling it? I would say that it gets initialized by providing the data. I would also like the name "build". Because the result is a function that builds stuff. That also fits with the buildStep terminology.

Reply:
Discussed in chat: We will rename initializedPlugin to build.